### ISSUE TITLE
Unblocking an index using 'blocks.read_only: false' doesn't work like 'index.blocks.read_only: false'

### ISSUE THREAD

**Issue by sethmlarson** [elastic/elasticsearch#80383]
User [reported a bug](https://discuss.elastic.co/t/python-client-unable-to-remove-read-only-block/288032/8?u=sethmlarson) that they weren't able to remove the `read_only` block from an index.

They were adding a block via:

```
PUT /index/_blocks/read_only
```

Then attempting to unblock the index via

```
PUT /index/_settings
{
  "blocks.read_only": false
}
```

and saw the following error:

```
index [index] blocked by: [FORBIDDEN/5/index read-only (api)];
```

However when you use `index.blocks.read_only` the unblock succeeds:

```
PUT /index/_settings
{
  "index.blocks.read_only": false
}
```

But it's allowed to set `read_only` using `blocks.read_only: true`

```
PUT /index/_settings
{
  "blocks.read_only": true
}
```

This asymmetry is either a bug or confusing, could this be fixed? I tested using 7.16-SNAPSHOT.

**Comment by DaveCTurner** [comment]
Definitely a bug, but this is already tracked at https://github.com/elastic/elasticsearch/issues/48325 so I'm closing this as a duplicate.
